### DISCUSSION TITLE
can matc/matinfo export glsl code with src comments intact?

### DISCUSSION THREAD

**Discussion by EnviralDesign** [google/filament#5387] [Q&A] [answered]
Right now I'm successfully exporting a .mat file through matc, then through matinfo to acquire the ascii shader code for vertex and fragment.

I find the comments scattered through out all the shader code in the github source really helpful, and would love if this could be included some how with the exported code.

Also side note, the glsl code that comes out of matinfo is modified so that there is exactly 1 space between every line, no indentation, etc. Are there any built in ways to change the way the code is formatted coming out of it?

**Comment by romainguy** [comment, marked as answer]
Shaders are stored in a special way to save on space, and that includes removing unnecessary spaces etc. They are also run through an optimizer by default. You can pass -g to match to preserve the original shaders as much as possible but it won't include comments as they are removed by the preprocessor.
